llvm/lib/IR/Instructions.cpp

```diff
@@ -918,6 +918,18 @@ LandingPadInst *InvokeInst::getLandingPadInst() const {
   return cast<LandingPadInst>(getUnwindDest()->getFirstNonPHI());
 }
 
+void InvokeInst::updateProfWeight(uint64_t S, uint64_t T) {
```

Comment:
should this be moved to CallBase?

Reply:
It makes sense to move them to `CallBase`. Will need to construct a test case (possibly based on https://github.com/llvm/llvm-project/blob/main/clang/test/CodeGen/asm-goto.c)
- The difference between 1) current patch and 2) moving this to 'CallBase' and calling `updateProfWeight` on a `CallBase` is that, branch weight of terminating instruction `callbr` ([lowered](https://llvm.org/docs/LangRef.html#callbr-instruction) from inline assembly [goto](https://gcc.gnu.org/onlinedocs/gcc/Extended-Asm.html#GotoLabels)) will also be updated.

Reply:
https://llvm.org/docs/BranchWeightMetadata.html#other mentions _Other terminator instructions are not allowed to contain Branch Weight Metadata_, indicating `callbr` instruction shouldn't have branch weight metadata attached.

Is there a reason why branch weight `!prof` cannot be attached to `callbr`?

Reply:
also from https://news.ycombinator.com/item?id=21018901, _Support for asm goto, enabling for example the mainline Linux kernel for x86_64 to build with Clang_

Reply:
> Is there a reason why branch weight !prof cannot be attached to callbr?

nvm seems doc is just stale according to IR verifier [code](https://github.com/llvm/llvm-project/blob/083d8aa03aca55b88098a91e41e41a8e321a5721/llvm/lib/IR/Verifier.cpp#L4758-L4759)